src/services/api/apiService.ts

```diff
@@ -0,0 +1,201 @@
+/**
+ * Facilitates http request creation by
+ * - offering helper methods with simplified interfaces
+ * - allowing to hook into before/after a request is made or if an error occurs
+ * - doing  preparatory work for network error request handling
+ */
+import ky, { Options as KyOptions } from 'ky';
+import config from '~/config';
+import store from '~/store';
+import { JSONValue, RequestOptions } from './types';
+import { selectors as UserStateSelectors } from '~/pages/User/UserState';
+import { ApiError, NetworkError, TimeoutError } from './httpErrors';
+import { emptyFunc } from '~/services/api/utils';
+
+// setup ky
+
+const configuredKy = ky.create({
+  prefixUrl: config.apiUrl,
+  hooks: {
+    beforeRequest: [
+      (req: Request) => {
+        const stateRoot = store.getState();
+        const token = UserStateSelectors.getToken(stateRoot);
+        if (token) {
+          req.headers.set('Authorization', `JWT ${token}`);
+        }
+      }
+    ]
+  }
+});
+
+// Generic Request Handler
+
+const defaultRequestOptions: RequestOptions = {
```

Comment:
The Syntax might be a bit tricky to read, I will try to explain what is going on here.

I want to 
* destructure the ´RequestOptions´ that `request()` takes as second arg
in order to assign default values and to have a bit less code in the function body
(in my eyes that is a good pattern)

and - at the same time -

* assign a default value to that arg, because it is optional, think of GET requests.

However you are right, I am assigning a default value to `setSubmitting()` [in the function body](https://github.com/FixMyBerlin/fixmy.frontend/pull/370/files#diff-7eedf2f3e1f7c249d3f4cdccfa29137fR52), which might seem a bit odd.
I just don't want to apply the same pattern (destructuring + default value) for `RequestOptions.callbacks` too, because IMHO that would make the declaration of arguments (even more?) unreadable.


****